engine/table/src/main/java/io/deephaven/engine/table/impl/QueryTable.java

```diff
@@ -732,24 +731,46 @@ public Table aggAllBy(AggSpec spec, ColumnName... groupByColumns) {
             }
             final List<ColumnName> groupByList = Arrays.asList(groupByColumns);
             final List<ColumnName> tableColumns = definition.getTypedColumnNames();
-            final Optional<Aggregation> agg = AggregateAllTable.singleAggregation(spec, groupByList, tableColumns);
+            final Optional<Aggregation> agg = singleAggregation(spec, groupByList, tableColumns);
             if (agg.isEmpty()) {
                 throw new IllegalArgumentException(
                         "aggAllBy has no columns to aggregate: spec=" + spec + ", groupByColumns="
                                 + toString(groupByList));
             }
-            final QueryTable tableToUse = (QueryTable) AggAllByUseTable.of(this, spec);
             final List<? extends Aggregation> aggs = List.of(agg.get());
             final MemoizedOperationKey aggKey = MemoizedOperationKey.aggBy(aggs, false, null, groupByList);
-            return tableToUse.memoizeResult(aggKey, () -> {
+            return memoizeResult(aggKey, () -> {
                 final QueryTable result =
-                        tableToUse.aggNoMemo(AggregationProcessor.forAggregation(aggs), false, null, groupByList);
+                        aggNoMemo(AggregationProcessor.forAggregation(aggs), false, null, groupByList);
                 spec.walk(new AggAllByCopyAttributes(this, result));
                 return result;
             });
         }
     }
 
+    /**
+     * Computes the single-aggregation from the agg-all implied by the {@code spec} and {@code groupByColumns} by
+     * removing the {@code groupByColumns} and any extra columns implied by the {@code spec}.
+     *
+     * @param spec the spec
+     * @param groupByColumns the group by columns
+     * @param tableColumns the table columns
+     * @return the aggregation, if non-empty
+     */
+    static Optional<Aggregation> singleAggregation(
```

Comment:
```suggestion
    @VisibleForTesting
    static Optional<Aggregation> singleAggregation(
```

Reply:
Will add this in my next PR to avoid another review cycle.